app/assets/javascripts/redux/modules/models/listenSignals.js

```diff
@@ -168,3 +190,16 @@ export const addListenSignalData = (payload) => {
     ],
   });
 };
+
+export const updateListenSignalData = (payload, signalId) => {
+  return createRequestAction({
+    endpoint: listenSignalEndpoint(signalId),
+    method: 'PUT',
+    body: JSON.stringify(payload),
+    types: [
+      { type: LISTEN_SIGNALS_PUT_REQUEST, meta: { signalId } },
```

Comment:
This will create a key `signalId` on the `meta` property of the action but in the `handleListenSignalFailResponse` we're expecting an `id` key. Change this and the argument to `id`.